BootloaderCommonPkg/Library/BootloaderCommonLib/BootloaderCommonLib.c

```diff
@@ -359,22 +351,67 @@ GetComponentInfoByPartition (
       //
       if ( ((EntryDesc.Flags & (FLASH_MAP_FLAGS_NON_REDUNDANT_REGION | FLASH_MAP_FLAGS_NON_VOLATILE_REGION)) != 0) ||
            (((IsBackupPartition ? FLASH_MAP_FLAGS_BACKUP : 0) ^ (EntryDesc.Flags & FLASH_MAP_FLAGS_BACKUP)) == 0) ) {
-        PcdBase = (UINT32) (RomBase + EntryDesc.Offset);
-        PcdSize = EntryDesc.Size;
+        CopyMem((VOID *)Entry, (VOID *)&EntryDesc, sizeof(FLASH_MAP_ENTRY_DESC));
         Status = EFI_SUCCESS;
-        break;
+        return Status;
       }
     }
   }
 
+  return Status;
+}
+
+/**
+  Gets component information from the flash map by partition.
+
+  This function will look for the component matching the input signature
+  in the flash map, if found, it will look for the component with back up
+  flag based on the backup partition parmeter and will return the
+  base address and size of the component.
+
+  @param[in]  Signature         Signature of the component information required
+  @param[in]  IsBackupPartition TRUE for Back up copy, FALSE for primary copy
+  @param[out] Base              Base address of the component
+  @param[out] Size              Size of the component
+
+  @retval    EFI_SUCCESS    Found the component with the matching signature.
+  @retval    EFI_NOT_FOUND  Component with the matching signature not found.
+
+**/
+EFI_STATUS
+EFIAPI
+GetComponentInfoByPartition (
+  IN  UINT32     Signature,
+  IN  BOOLEAN    IsBackupPartition,
+  OUT UINT32     *Base,
+  OUT UINT32     *Size
+  )
+{
+  EFI_STATUS            Status;
+  FLASH_MAP_ENTRY_DESC  Entry;
+  FLASH_MAP             *FlashMapPtr;
+  UINT32                RomBase;
+
+  Status = GetComponentEntryByPartition(Signature, TRUE, &Entry);
+  if ((Status == EFI_NOT_FOUND) || (Status == EFI_UNSUPPORTED)) {
+    return Status;

```

Comment:
Better use EFI_ERROR (Status), so that when the impl of GetComponentEntryByPartition is changed to return other values, this check still works.

Reply:
Done